src/Shared.Core/Utilities/OpenFileDialog.cs

```diff
@@ -66,7 +73,10 @@ public static string[] ShowDialog(string title, string initialDir, string filter
             ofn.filter = filter.Replace("|", "\0") + "\0";
             ofn.fileTitle = new String(new char[MAX_FILE_LENGTH]);
             ofn.maxFileTitle = ofn.fileTitle.Length;
-            ofn.initialDir = initialDir;
+            if (LastOpenedPaths.TryGetValue(initialDir, out string lastOpenedPath) && KoikatuAPI.RememberFilePickerFolder.Value)
+                ofn.initialDir = lastOpenedPath;
+            else
+                ofn.initialDir = initialDir;
```

Comment:
Probably better to just check if lastOpenedPath is null or empty, and check the setting first.